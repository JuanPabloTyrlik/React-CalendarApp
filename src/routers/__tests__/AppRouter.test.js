import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { AppRouter } from '../AppRouter';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Tests on <AppRouter />', () => {
    test('should render "Loading..."', () => {
        const store = mockStore({
            auth: {
                checking: true,
                uid: '123',
            },
        });

        const wrapper = mount(
            <Provider store={store}>
                <AppRouter />
            </Provider>
        );
        expect(wrapper).toMatchSnapshot();
    });
    test('should render Public Route', () => {
        const store = mockStore({
            auth: {
                checking: false,
            },
        });

        const wrapper = mount(
            <Provider store={store}>
                <AppRouter />
            </Provider>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.login-container').exists()).toBe(true);
    });
    test('should render Private Route', () => {
        const store = mockStore({
            auth: {
                checking: false,
                uid: '123',
            },
            calendar: {
                events: [],
                activeEvent: null,
            },
            ui: {
                modalOpen: false,
            },
        });

        const wrapper = mount(
            <Provider store={store}>
                <AppRouter />
            </Provider>
        );

        expect(wrapper.find('.calendar-screen').exists()).toBe(true);
    });
});
