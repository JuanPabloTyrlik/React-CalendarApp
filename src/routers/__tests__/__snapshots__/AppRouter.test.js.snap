// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <AppRouter /> should render "Loading..." 1`] = `
<h5>
  Loading...
</h5>
`;

exports[`Tests on <AppRouter /> should render Public Route 1`] = `
<div>
  <div
    className="container login-container"
  >
    <div
      className="row"
    >
      <div
        className="col-md-6 login-form-1"
      >
        <h3>
          Ingreso
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="Lemail"
              onChange={[Function]}
              placeholder="Correo"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="Lpassword"
              onChange={[Function]}
              placeholder="Contraseña"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="btnSubmit"
              type="submit"
              value="Login"
            />
          </div>
        </form>
      </div>
      <div
        className="col-md-6 login-form-2"
      >
        <h3>
          Registro
        </h3>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="Rname"
              onChange={[Function]}
              placeholder="Nombre"
              type="text"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="Remail"
              onChange={[Function]}
              placeholder="Correo"
              type="email"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="Rpassword"
              onChange={[Function]}
              placeholder="Contraseña"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              name="Rpassword2"
              onChange={[Function]}
              placeholder="Repita la contraseña"
              type="password"
              value=""
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="btnSubmit"
              type="submit"
              value="Crear cuenta"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
